pkg/storage/ingest/fetcher.go

```diff
@@ -657,10 +679,24 @@ func (r *concurrentFetchers) start(ctx context.Context, startOffset int64, concu
 				continue
 			}
 			nextFetch = nextFetch.UpdateBytesPerRecord(result.fetchedBytes, len(result.Records))
+
+			// We have some ordered records ready to be sent to PollFetches(). We store the fetch
+			// result in bufferedResult, and we flip readyBufferedResults to the channel used by
+			// PollFetches().
 			bufferedResult = result
 			readyBufferedResults = r.orderedFetches
 
+			// We increase the count of buffered records only for ordered records that we're sure
+			// will not be discarded later, so that we can get an accurate tracking of the records
+			// ready to be polled by PollFetches() but not polled yet.
+			r.bufferedFetchedRecords.Add(int64(len(result.Records)))
```

Comment:
it's more accurate if we track this in `.run()`. Technically the single fetcher (`run()`) can be blocked and trying to send its records to the result channel, but the `start()` goroutine is waiting to send to `orderedFetches`. In that case the records buffered within the `run()` are not counted.

I even think that most of the records will be buffered there because `bufferedResult` is only one fetch result.